src/Core/Rendering/RenderController.cpp

```diff
@@ -537,6 +538,62 @@ namespace MxEngine
         this->RenderToTexture(output, shader);
     }
 
+    void RenderController::ApplyGodRayEffect(CameraUnit& camera, TextureHandle& input, TextureHandle& output)
+    {
+        if (!camera.Effects->IsGodRayEnabled())
+            return;
+        int dirLightCount = this->Pipeline.Lighting.DirectionalLights.size();
+        if (dirLightCount <= 0)
```

Comment:
minor: size_t is always `>= 0`, so here you can just check `== 0`

Reply:
also dont convert size_to to int, (`int dirLightCount = this->Pipeline.Lighting.DirectionalLights.size();`